### PR TITLE
Move getStaticPaths call to happen during generation

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- The call to `getStaticPaths` during the build is moved to the end of the process, during page generation.
- This prevents us from ever needing to call `vite.ssrLoadModule` anywhere in the build now. We are 100% completely static.
- This should improve performance as we are no longer double-building this code.
- This also fixes a bug with Lit and `getStaticPaths`. Before this change if you use a Lit element in a page behind getStaticPaths you would get a `window is undefined` error. This was because getStaticPaths was being called upfront, when the renderer had not yet run, and therefore the shim was not applied. Now it happens as it should.
- Closes https://github.com/withastro/astro/issues/3951

## Testing

- Tests added

## Docs

N/A, bug fix.

### CODE REVIEW

File: packages/astro/src/core/build/page-data.ts [hunk 1]
```diff
@@ -133,16 +89,3 @@ export async function collectPagesData(
...
 }
-
-async function getStaticPathsForRoute(
-	opts: CollectPagesDataOptions,
-	route: RouteData
-): Promise<RouteCacheEntry> {
-	const { astroConfig, logging, routeCache, ssr, viteServer } = opts;
-	if (!viteServer) throw new Error(`vite.createServer() not called!`);
-	const filePath = new URL(`./${route.component}`, astroConfig.root);
-	const mod = (await viteServer.ssrLoadModule(fileURLToPath(filePath))) as ComponentInstance;
```
Comment (matthewp) [inline packages/astro/src/core/build/page-data.ts:144]: this was the last `ssrLoadModule` call still happening during the build.

**Review by FredKSchott — APPROVED** [review]
Nice! I know you've been working towards this for a while. LGTM! If tests are passing, I'll merge this to get it in in time for the nightly release.

**Comment by FredKSchott** [comment]
Ah, I'd hoped the breaking test was a small bug but it looks like its due to an actual regression. Probably due to a side-effect of changing the order that you build files in the `build` command. Reverted my attempt at the small fix, and will leave for you to solve!

**Comment by matthewp** [comment]
Yeah, I think it's because `manifest.routes` is ordered and that is how we were previously adding paths and this PR changes that. I think I know how to fix it though.

**Pull request merged**